tiny_dnn/layers/cell.h

```diff
@@ -0,0 +1,46 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+#include "tiny_dnn/core/backend.h"
+#include "tiny_dnn/core/framework/device.fwd.h"
+
+namespace tiny_dnn {
+
+/**
+ * Abstract class for recurrent cells.
+ **/
+class cell: public layer {
+ public :
+
+  cell(core::backend_t backend_type) : backend_type_(backend_type), layer({}, {}) {}
```

Comment:
I think that also for convention we should specify the backend type in the recurrent_layer